src/browser-frontend/sagas/webcontents-action-handlers.js

```diff
@@ -53,6 +54,12 @@ function* onPageDidStartLoading({ payload: { pageId } }) {
 function* onPageDidStopLoading({ payload: { pageId } }) {
   yield put(PagesModelActions.setPageLoadState({
     pageId,
+    loadState: DomainPageMetaModel.LOAD_STATES.ANIMATED,
+  }));
+
+  yield call(delay, 400);
+  yield put(PagesModelActions.setPageLoadState({
+    pageId,
     loadState: DomainPageMetaModel.LOAD_STATES.LOADED,
   }));
```

Comment:
Conflating UI state with domain model state means there's something very fishy about this approach. Page load states should mirror their corresponding HTTP counterparts, and we shouldn't just mark a page as "loaded" 400ms later than it actually happens.